deploy/v2beta1/mpi-operator.yaml

```diff
@@ -7942,6 +7942,10 @@ spec:
                 description: SSHAuthMountPath is the directory where SSH keys are
                   mounted. Defaults to "/root/.ssh".
                 type: string
+              waitForWorkers:
```

Comment:
who's waiting?

Perhaps a better API is `launcherCreationPolicy: AtStartup | WaitForWorkersReady`, but I'm open to other suggestions.

Reply:
I prefer `launcherCreationPolicy: AtStartup | WaitForWorkersReady` rather than `launcherStartup: OnCreation|AfterWorkersReady`.

Reply:
So, do you insist on changing the name and values?

Reply:
+1 on @tenzen-y's proposal

Reply:
> So, do you insist on changing the name and values?

Yes, I do. The API can not be changed easily. So we need to decide APIs carefully.

Reply:
Can I somehow just regenerate the name? I did everything manually and it is a bit pain to rename it this way..

Reply:
You can use `make generate`: https://github.com/kubeflow/mpi-operator/blob/21f326d1d2e5f1beed667da0cc61e1d07158ef0c/Makefile#L84-L89

Reply:
1. Modify https://github.com/kubeflow/mpi-operator/blob/21f326d1d2e5f1beed667da0cc61e1d07158ef0c/pkg/apis/kubeflow/v2beta1/types.go
2. Modify controller and test
3. Run `make generate`

Reply:
I updated the above steps.

Reply:
Should it work?

```
Generating V2 OpenAPI specification ...
API rule violation: list_type_missing,github.com/kubeflow/common/pkg/apis/common/v1,JobStatus,Conditions
Generating V2 swagger file ...
# command-line-arguments
hack/python-sdk/main.go:48:23: undefined: v2beta1.GetOpenAPIDefinitions
make: *** [Makefile:87: generate] Error 2
```

Reply:
@xhejtman On my local, it worked fine. Let me check your changes. So can you push WIP commit into this PR?

Reply:
pushed

Reply:
I could generate codes using your latest commit on my local.
Checking your comment, it seems that you failed to generate or pick `pkg/apis/kubeflow/v2beat1/openapi_generated.go`.

> Should it work?
> 
> Generating V2 OpenAPI specification ...
> API rule violation: list_type_missing,github.com/kubeflow/common/pkg/apis/common/v1,JobStatus,Conditions
> Generating V2 swagger file ...
> # command-line-arguments
> hack/python-sdk/main.go:48:23: undefined: v2beta1.GetOpenAPIDefinitions
> make: *** [Makefile:87: generate] Error 2

@xhejtman Where did you put the mpi-operator repository on your local? Do you have this repo under the GOPATH?

Reply:
/tmp/mpi-operator, just ad-hoc dir to test.

Reply:
@xhejtman Can you put this repo under the GOPATH?

Reply:
Then, you will be able to generate codes.

Reply:
I really don't know what to do. I exported GOPATH=/tmp/mpi-operator. `make generate` fails. Unset gopath, it alternates between two failures.
```
xhejtman@osiris:/tmp/mpi-operator$ export GOPATH=/tmp/mpi-operator
xhejtman@osiris:/tmp/mpi-operator$ make generate
go generate ./pkg/... ./cmd/...
$GOPATH/go.mod exists but should not
make: *** [Makefile:84: generate] Error 1
xhejtman@osiris:/tmp/mpi-operator$
```

```
make generate
go generate ./pkg/... ./cmd/...
hack/update-codegen.sh
/tmp/mpi-operator /tmp/mpi-operator
/tmp/mpi-operator
Generating deepcopy funcs
Generating clientset for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/clientset
Generating listers for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/listers
Generating informers for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/informers
make manifest
make[1]: Entering directory '/tmp/mpi-operator'
mkdir -p /tmp/mpi-operator/bin
/tmp/mpi-operator/bin/controller-gen "crd:generateEmbeddedObjectMeta=true" paths="./..." output:crd:artifacts:config=manifests/base
hack/generate-manifest.sh /tmp/mpi-operator/bin/kustomize
make[1]: Leaving directory '/tmp/mpi-operator'
hack/python-sdk/gen-sdk.sh
Generating V2 OpenAPI specification ...
API rule violation: list_type_missing,github.com/kubeflow/common/pkg/apis/common/v1,JobStatus,Conditions
Generating V2 swagger file ...
# command-line-arguments
hack/python-sdk/main.go:48:23: undefined: v2beta1.GetOpenAPIDefinitions
make: *** [Makefile:87: generate] Error 2
```

```
xhejtman@osiris:/tmp/mpi-operator$ make generate
go generate ./pkg/... ./cmd/...
hack/update-codegen.sh
/tmp/mpi-operator /tmp/mpi-operator
/tmp/mpi-operator
Generating deepcopy funcs
Generating clientset for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/clientset
Generating listers for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/listers
Generating informers for kubeflow:v2beta1 at github.com/kubeflow/mpi-operator/pkg/client/informers
make manifest
make[1]: Entering directory '/tmp/mpi-operator'
mkdir -p /tmp/mpi-operator/bin
/tmp/mpi-operator/bin/controller-gen "crd:generateEmbeddedObjectMeta=true" paths="./..." output:crd:artifacts:config=manifests/base
hack/generate-manifest.sh /tmp/mpi-operator/bin/kustomize
make[1]: Leaving directory '/tmp/mpi-operator'
hack/python-sdk/gen-sdk.sh
mv: cannot stat 'pkg/apis/kubeflow/v2beta1/openapi_generated.go': No such file or directory
make: *** [Makefile:87: generate] Error 1
```

Reply:
@xhejtman We need to follow the Golang directory structure:

https://go.dev/doc/gopath_code#GOPATH

Reply:
I fixed it manually without generate as I do not know how to do it with gopath